engine/table/src/main/java/io/deephaven/engine/util/LayoutHintBuilder.java

```diff
@@ -278,6 +343,56 @@ public LayoutHintBuilder hide(Collection<String> cols) {
         return this;
     }
 
+    /**
+     * @see LayoutHintBuilder#group(String, List, Color)
+     */
+    @ScriptApi
+    public LayoutHintBuilder group(String name, List<String> children) {
+        return group(name, children, (Color) null);
+    }
+
+    /**
+     * @see LayoutHintBuilder#group(String, List, Color)
+     */
+    @ScriptApi
+    public LayoutHintBuilder group(String name, List<String> children, String color) {
+        if (color == null || color.length() == 0) {
+            return group(name, children, (Color) null);
+        }
+        return group(name, children, new Color(color));
+    }
+
+    /**
+     * Create a named group of columns in the UI
+     *
+     * @param name the column group name. Must be a valid Deephaven column name
+     * @param children the columns and other groups belonging to this group
+     * @param color the background color for the group in the UI
+     * @return this LayoutHintBuilder
+     */
+    @ScriptApi
+    public LayoutHintBuilder group(String name, List<String> children, Color color) {
+        if (columnGroups == null) {
+            columnGroups = new HashMap<>();
```

Comment:
does order matter here, of the named groups? should this be a LinkedHashMap instead to keep that order?

Reply:
I'm not sure if order will matter yet (haven't fully gotten into the UI side of determining how these will be ordered when a user specifies columns 1, 2, 3 in 1 group and 0, 4, 5 in another). We might need the definition order so I'll switch to `LinkedHashMap` to maintain it